Add support for in-memory compression

lukevp opened the pull request:
Some use cases of this middleware may require the ability to compress content in-memory as either a string or a byte array (Uint8Array).  One example would be when using this library to compress a templated file (eg. read in a file from disk, inject template variables, and then return it compressed).  The way this library is currently implemented, only static content on disk can be compressed, although it's straightforward to support other patterns of delivering the underlying content to the library.


~~NOTE: I have Deno set up in my VS Code with the Deno Lint settings, which automatically added semicolons and converted apostrophes to quotes, among other formatting changes.  My preference is to use the standard auto-formatting tools provided by a language (and even commit this config to the repo) so that everyone's code is formatted the same.  If this bothers you, I can revert the formatting changes and attempt to make a compatible deno lint configuration that matches this coding style, although I admittedly don't have much experience with `deno lint` and typically work with `prettier` and `eslint`.~~
**PR updated with prettier settings to match project settings**

lukevp committed:
Update example app with console usage message

lukevp committed:
Reformat example file with deno lint

lukevp committed:
Add support for in-memory compression

Allow passing in bodyText or bodyBinary in addition to path.
Only 1 will be allowed.
The options type was made as a union type to allow existing code to be used unchanged so there was no breaking change in the API for end users.

lukevp committed:
Add test cases for in memory compression

lukevp committed:
Reformat all files with deno fmt

talentlessguy commented:
could you please remove `deno fmt` changes and follow Standard.js / prettier w/o semi and single quote? thx

lukevp committed:
Adjust settings for prettier and create prettier config

PR Feedback to use `prettier` for formatting instead of `deno fmt`
Create prettierrc file so prettier settings can be applied to project

lukevp committed:
Revert `deno fmt` formatting changes in README

lukevp committed:
Remove trailing comma setting to bring code format more in line with previous style

lukevp committed:
Update example with prettier format settings

lukevp committed:
Update line width to match the wrapping that was in the original project

lukevp commented:
@talentlessguy  Thanks for the feedback, ready for re-review.  Included a .prettierrc with the appropriate settings and used prettier to format instead of `deno fmt`.

talentlessguy commented:
could you please remove the .prettierrc file?

Commit:
Delete .prettierrc

talentlessguy commented:
tests seem to fail still, @lukevp

lukevp committed:
Update file sizes to use the unix newline sizes

This was originally tested on Windows so the file sizes differed due to carriage returns

lukevp commented:
Tests failed because I'm on Windows with CR/LF line endings on the file, and I used the Windows file sizes in the test cases.  Should really use the unix style for this so I updated the test cases.

Any reason you don't want the .prettierrc file in the project?  It lets new contributors inherit the prettier style you've chosen for the project automatically when using vs code.  I typically put a prettier and eslint config in each project along with a .gitattributes file that forces unix style line endings.  Wanted to get your perspective on this so I understand the reason, but of course I'm 
 happy to contribute however you prefer as it's your project :)

talentlessguy commented:
@lukevp because .prettierrc will require users to have Node.js

I wonder if it makes sense to add editorconfig, although it doesn't support configuring quotes and semicolons

maybe dprint.dev would be a better thing

anyways it's not related to this PR, better address this in another issue
